storybook/public/service-worker.js

```diff
@@ -0,0 +1,38 @@
+// Add install handler.
+self.addEventListener( 'install', ( event ) => {
+	// Force activation immediately.
+	event.waitUntil( self.skipWaiting() );
+} );
+
+// Add activate handler.
+self.addEventListener( 'activate', ( event ) => {
+	// Take control immediately.
+	event.waitUntil( self.clients.claim() );
+} );
+
+// Add fetch handler.
+self.addEventListener( 'fetch', ( event ) => {
+	const url = new URL( event.request.url );
+
+	// Check if this is the Google Sign-In script.
```

Comment:
I've hardwired a check for the SiWG script for the PoC, but this could be adapted to be config-driven for easier reuse.